web/src/peer_data.ts

```diff
@@ -244,14 +244,20 @@ export function rewire_get_subscriber_count(value: typeof get_subscriber_count):
 }
 
 export function get_subscribers(stream_id: number): number[] {
+    if (!fetched_stream_ids.has(stream_id)) {
+        blueslip.error("Getting subscribers for stream without full subscriber data", {
+            stream_id,
+        });
```

Comment:
Probably you can add a comment that we don't raise an exception because we'd rather not break the caller, but this should not occur. I think there's only two callers, so another option would be to audit those callers and make this an assert.

Hmm, it's not obvious that the call from `get_filtered_user_id_list` actually would pass that assert. Needs investigation.

Reply:
Might be useful to rename to `get_loaded_subscriber_ids` anyway, or `get_subscriber_ids_assert_loaded`, depending if we're including the assert.

Reply:
> we don't raise an exception because we'd rather not break the caller, but this should not occur

Isn't this true of any use of `blueslip.error`? Or are there situations where we'd expect blueslip errors to occur?

> Hmm, it's not obvious that the call from `get_filtered_user_id_list` actually would pass that assert. Needs investigation.

This is related to your above comment -- if all visible users in the buddy list are users we know about (which I think really should be the case, so if it's not we should adjust what we're fetching), then `get_filtered_user_id_list` would pass the assert. Notably, we only show _all_ users for larger organizations when we're searching users, and there's a wait to fetch full subscriber data before fetching search results (#34677).

Reply:
> Or are there situations where we'd expect blueslip errors to occur?

If we fail to understand how to code works :). The trade-off being discussed is `throw new Error/assert`, which, if the unexpected happens, results in the rest of the function not running. I am suggesting our certainty is probably not high enough to want to do that.

Reply:
Right, that's why I did that here. But my point is that **anytime** we're using `blueslip.error` and not actively throwing an error, it would presumably be a situation where "we don't raise an exception because we'd rather not break the caller, but this should not occur" -- so I'm not sure if adding a comment about that is necessary.

Reply:
Ahh, yeah, I was thinking we might want to replace this with an assert after it's been in production for a couple weeks, if we believe it shouldn't be possible, but don't have a short proof that would justify making it an assert now.

Reply:
I added this:

```
    // TODO: Convert this to an `assert` once we can be more sure that this never
    // happens.
```

Reply:
Works for me.